storage/src/main/java/org/apache/kafka/storage/internals/log/UnifiedLog.java

```diff
@@ -2416,15 +2416,15 @@ public static void rebuildProducerState(ProducerStateManager producerStateManage
                                             Time time,
                                             boolean reloadFromCleanShutdown,
                                             String logPrefix) throws IOException {
-        List<Optional<Long>> offsetsToSnapshot = new ArrayList<>();
+        List<Long> offsetsToSnapshot = new ArrayList<>();
         if (segments.nonEmpty()) {
```

Comment:
Could you please consider using `segments.lastSegment().ifPresent` to rewrite it?